src/conversions/chrono.rs

```diff
@@ -0,0 +1,848 @@
+        let ms = dt.get_fold() as u32 * 1_000_000 + dt.get_microsecond();
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
+            tzinfo.extract()?
+        } else {
+            return Err(PyTypeError::new_err("Not datetime.timezone.tzinfo"));
+        };
+        let dt = NaiveDateTime::new(
+            NaiveDate::from_ymd(dt.get_year(), dt.get_month().into(), dt.get_day().into()),
+            NaiveTime::from_hms_micro(h, m, s, ms),
+        );
+        Ok(DateTime::from_utc(dt, tz))
+    }
+}
+
+impl FromPyObject<'_> for DateTime<Utc> {
+    fn extract(ob: &PyAny) -> PyResult<DateTime<Utc>> {
+        let dt = <PyDateTime as PyTryFrom>::try_from(ob)?;
+        let ms = dt.get_fold() as u32 * 1_000_000 + dt.get_microsecond();
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
+            tzinfo.extract()?
+        } else {
+            return Err(PyTypeError::new_err("Not datetime.timezone.utc"));
+        };
+        let dt = NaiveDateTime::new(
+            NaiveDate::from_ymd(dt.get_year(), dt.get_month().into(), dt.get_day().into()),
+            NaiveTime::from_hms_micro(h, m, s, ms),
+        );
+        Ok(DateTime::from_utc(dt, tz))
+    }
+}
+
+// Utiliy function used to convert PyDelta to timezone
+fn pytimezone_fromoffset<'a>(py: &Python<'a>, td: &PyDelta) -> &'a PyAny {
+    // Safety: py.from_borrowed_ptr needs the cast to be valid.
+    // Since we are forcing a &PyDelta as input, the cast should always be valid.
+    unsafe {
+        PyDateTime_IMPORT();
+        py.from_borrowed_ptr(PyTimeZone_FromOffset(td.as_ptr()))
+    }
+}
+
+impl ToPyObject for FixedOffset {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        let seconds_offset = self.local_minus_utc();
+        // XXX: Here we don't normalize, otherwise the meaning
+        // of the offset changes. Normalizing a negative PyDelta and converting
+        // back to rust would transform a `-00:00:01` offset into `23:59:59`.
+        let td =
+            PyDelta::new(py, 0, seconds_offset, 0, false).expect("Failed to contruct timedelta");
+        pytimezone_fromoffset(&py, td).into()
+    }
+}
+
+impl IntoPy<PyObject> for FixedOffset {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for FixedOffset {
+    /// Convert python tzinfo to rust [`FixedOffset`].
+    ///
+    /// Note that the conversion will result in precision lost in microseconds as chrono offset
+    /// does not supports microseconds.
+    fn extract(ob: &PyAny) -> PyResult<FixedOffset> {
+        let py_tzinfo = <PyTzInfo as PyTryFrom>::try_from(ob)?;
+        let py_timedelta = py_tzinfo.call_method1("utcoffset", (ob.py().None(),))?;
```

Comment:
I'm not super familiar with the `utcoffset` method, in the Python docs it looks like it takes a datetime object as argument, but we're passing `None` here. Can you clarify how this works?

Is it also potentially brittle? E.g. the offset might change depending on the date if it's an IANA timezone?

Reply:
Passing `None` here only works for tzinfo that represents a fixed offset in Python, so it should be correct since that's what we are converting here. You are right, this would fail for any other kind of tzinfo, since the delta would change depending on the date.

See this example: https://docs.python.org/3/library/datetime.html?highlight=utcoffset#examples-of-usage-time
And this class: https://docs.python.org/3/library/datetime.html?highlight=utcoffset#datetime.timezone

The most notable example is UTC:
```python
from datetime import timezone, timedelta
timezone.utc.utcoffset(None) == timedelta(0)
```

But it works for any datetime with tzinfo of class `datetime.timezone` (which is just three UTCs in a trenchcoat):
```python
from datetime import timezone, timedelta, datetime
datetime.now(tz=timezone(timedelta(hours=2))).tzinfo.utcoffset(None) == timedelta(seconds=7200)
```

Reply:
Great, so it looks like `None` for `zoneinfo.ZoneInfo` returns `None`?

```python
>>> zoneinfo.ZoneInfo("Europe/London").utcoffset(None) == None
True
```

Three final suggestions here:
- Can we please add a comment here documenting that `None` will only work for fixed offsets (as it's not immediately clear to readers IMO).
- Please add a test confirming that `zoneinfo.ZoneInfo("Europe/London")` doesn't convert to `FixedOffset` (so that refactoring later doesn't break this design).
- Maybe add an improved error message here? e.g. `format!("`{:?}` is not a fixed offset timezone", py_tzinfo.repr().unwrap_or_else("repr failed"));` (I think that won't exactly compile but should give you the idea.)